cmd/compose/compose.go

```diff
@@ -201,14 +188,38 @@ func (o *ProjectOptions) toProjectName() (string, error) {
 		return envProjectName, nil
 	}
 
-	project, err := o.ToProject(nil)
+	project, err := o.ToProject(dockerCli, nil)
 	if err != nil {
 		return "", err
 	}
 	return project.Name, nil
 }
 
-func (o *ProjectOptions) ToProject(services []string, po ...cli.ProjectOptionsFn) (*types.Project, error) {
+func (o *ProjectOptions) ToProject(dockerCli command.Cli, services []string, po ...cli.ProjectOptionsFn) (*types.Project, error) {
```

Comment:
This gets used as part of shell completion, which ~probably~ definitely shouldn't try to clone Git repos to handle auto-complete.

What do you think about having a `ToProjectOffline` (which does _not_ take the `dockerCli` / attempt to create loaders), which could be used by `ValidArgsFunction`?

(If it returns an error since the project requires remote resources, shell completion will not return any results, which seems fine)

Reply:
Yeah, if the "offline" mode could use cache, that'd be pretty neat, but for simplicity, it might be fine to just ignore for the moment. My real concern is hitting `<tab>` and then having Compose start cloning a huge monorepo in the background by surprise 😬